### PR TITLE
[Improvement] making listening to event configurable

### PR DESCRIPTION
**Pull request by gabrielefantini**
Configurable event handling 
Tests refactored
Readme updated and fixed

### CODE REVIEW

File: observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/client/blindata/BDClientImpl.java [hunk 1]
```diff
@@ -126,12 +126,14 @@ public BDProductPortAssetsRes createDataProductAssets(BDProductPortAssetsRes dat
...
 
     @Override
-    public Optional<BDStewardshipResponsibilityRes> getResponsibility(String userUuid, String resourceIdentifier) throws BlindataClientException, BlindataClientResourceMappingException {
+    public Optional<BDStewardshipResponsibilityRes> getActiveResponsibility(String userUuid, String resourceIdentifier) throws BlindataClientException, BlindataClientResourceMappingException {
         try {
             BDStewardshipResponsibilitySearchOptions filters = new BDStewardshipResponsibilitySearchOptions();
             filters.setResourceIdentifier(resourceIdentifier);
             filters.setUserUuid(userUuid);
             filters.setRoleUuid(credentials.getRoleUuid());
+            filters.setEndDateIsNull(true);
```
Comment (gabrielefantini) [inline observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/client/blindata/BDClientImpl.java:135]: Gets only active responsibilities

File: observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/resources/odm/EventNotificationMapper.java
Comment (gabrielefantini) [inline observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/resources/odm/EventNotificationMapper.java:1]: Decoupled from ODM-platform

File: observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/services/DataProductPortAssetAnalyzer.java [hunk 2]
```diff
@@ -67,9 +67,12 @@ public List<BDDataProductPortAssetDetailRes> extractPhysicalResourcesFromPorts(L
                 }
 
                 PortStandardDefinition portStandardDefinition = externalComponentToPortStandardDefinition(externalComponentResources.get());
-                logger.info("Data product port: {} has specification: {} of version: {}", port.getFullyQualifiedName(), portStandardDefinition.getSpecification(), portStandardDefinition.getSpecificationVersion());
 
+                if (portIsNotValid(port, portStandardDefinition)) continue;
```
Comment (gabrielefantini) [inline observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/services/DataProductPortAssetAnalyzer.java:71]: Avoids ports that don't have specification defined

File: observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/services/NotificationEventConsumerService.java [hunk 3]
```diff
@@ -12,39 +17,41 @@
...
-        }
+        OBEventNotificationResource obEventNotificationResource = mapper.toObserverResource(eventNotificationResource);
+        self.consumeEventNotificationAsync(obEventNotificationResource);
+        eventNotificationResource.setStatus(EventNotificationStatus.PROCESSING);
         return eventNotificationResource;
     }
 
-    private EventNotificationResource updateEventNotificationOnNotificationService(
+    @Async
+    public void consumeEventNotificationAsync(OBEventNotificationResource eventNotificationResource) {
```
Comment (gabrielefantini) [inline observer-blindata-server/src/main/java/org/opendatamesh/platform/up/metaservice/blindata/services/NotificationEventConsumerService.java:42]: now the notification is received and the called immediately released

**Review by aliprax — APPROVED** [review]

**Pull request merged**